File: source/code/vbnet/authentication/create_directory_fb.vb
Imports Stormpath.SDK.Provider

Dim facebookDirectory = client.Instantiate(Of IDirectory)()
facebookDirectory.SetName("My Facebook Directory")

Await client.CreateDirectoryAsync(
    facebookDirectory,
    Function(options) options.ForProvider(client _
                                          .Providers() _
                                          .Facebook() _
                                          .Builder() _
                                          .SetClientId("YOUR_FACEBOOK_APP_ID") _
                                          .SetClientSecret("YOUR_FACEBOOK_APP_SECRET") _
                                          .Build()))

File: source/code/nodejs/authentication/create_directory_fb.js
var facebookDirectory = {
  name : 'my-facebook-directory',
  description : 'A Facebook directory',
  provider: {
    providerId: 'facebook',
    clientId: 'YOUR_FACEBOOK_APP_ID',
    clientSecret: 'YOUR_FACEBOOK_APP_SECRET'
  }
};

client.createDirectory(facebookDirectory, function (err, directory) {
  if (err) {
    return console.error(err);
  }

  console.log('Facebook directory created!');
});